components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1007,6 +1014,210 @@ public void updateResidentIdP(IdentityProvider identityProvider, String tenantDo
+                    }
+                    break;
+            }
+            return sortBy;
+        }
+    }
+
+    /**
+     * Validate sortOrder.
+     *
+     * @param sortOrder sortOrder ASC/DESC.
+     * @return Validated sortOrder and sortBy.
+     */
+    private String validateSortOrder(String sortOrder) {
+
+        if (StringUtils.isBlank(sortOrder)) {
+            sortOrder = IdPManagementConstants.DEFAULT_SORT_ORDER;
+            if (log.isDebugEnabled()) {
+                log.debug("sortOrder is empty. Therefore we set the default sortOrder value as ASC. SortOrder: " +
+                        sortOrder);
+            }
+        } else if (sortOrder.equals(IdPManagementConstants.DESC_SORT_ORDER)) {
+            sortOrder = IdPManagementConstants.DESC_SORT_ORDER;
+        } else {
+            sortOrder = IdPManagementConstants.DEFAULT_SORT_ORDER;
+            if (log.isDebugEnabled()) {
+                log.debug("sortOrder is incorrect. Therefore we set the default sortOrder value as ASC. SortOrder: "
+                        + sortOrder);
+            }
+        }
+        return sortOrder;
+    }
+
+    /**
+     * Validate limit.
+     *
+     * @param limit given limit value.
+     * @return validated limit and offset value.
+     */
+    private int validateLimit(int limit) {
+
+        if (limit > IdPManagementConstants.MAXIMUM_LIMIT_PER_PAGE) {
+            try {
+                String itemsPerPagePropertyValue = ServerConfiguration.getInstance()
+                        .getFirstProperty(IdPManagementConstants.ITEMS_PER_PAGE_PROPERTY);
+                if (log.isDebugEnabled()) {
+                    log.debug("Given limit exceed the maximum limit. Therefore we get the default limit from " +
+                            "carbon.xml. limit: " + limit);
+                }
+                if (StringUtils.isNotBlank(itemsPerPagePropertyValue)) {
+                    limit = Integer.parseInt(itemsPerPagePropertyValue);
+                } else {
+                    limit = IdPManagementConstants.DEFAULT_RESULTS_PER_PAGE;
+                    if (log.isDebugEnabled()) {
+                        log.debug("limit is incorrect. Therefore we set the default limit. limit:" + limit);
+                    }
+                }
+            } catch (NumberFormatException e) {
+                limit = IdPManagementConstants.DEFAULT_RESULTS_PER_PAGE;
+                log.warn("Error occurred while parsing the 'ItemsPerPage' property value in carbon.xml.", e);
+            }
+        }
+        return limit;
+    }
+
+    /**
+     * Validate offset.
+     *
+     * @param offset given offset value.
+     * @return validated limit and offset value.
+     */
+    private int validateOffset(int offset) {
+
+        if (offset < 0) {
+            offset = 0;
+            if (log.isDebugEnabled()) {
+                log.debug("Invalid offset applied. Therefore we set the default offset value as 0. offSet: " + offset);
```

Comment:
shouldn't we throw an error if invalid/negative offset is given?

Reply:
we shouldn't  throw an error for offset, we display the first page know when the offset value is negative.

Reply:
I think this is an error scenario. Let's throw and error.